### PR TITLE
Update quickstart and actor guide with component examples

### PR DESCRIPTION
**Pull request by brooksmtownsend**
## Feature or Problem
This PR updates our quickstart with Rust and TinyGo component examples. It also formalizes the use of `<Tabs>` groups/query strings around the documentation so that users who select one language/os/environment will continue to see that everywhere possible.

I also included a WASI reference which includes all imports/exports that we support in the host as generated by `wit-bindgen`. Unfortunately some are not shown here, like `wasi:cli`, but that's handled by wasmtime which isn't in our wit world. We may consider adding those explicitly to our worlds in the future.

## Related Issues
Requires https://github.com/wasmCloud/wasmCloud/pull/1266 to fully let the examples be easy to add keyvalue/logging

### Manual Verification
I ran through these examples through "adding additional functionality". This is a draft as I still need to:

- [x] Add query string parsing to the tinygo component
- [x] Add the keyvalue example
- [x] Update the actor developer guide to refer to components

### CODE REVIEW

File: docs/developer/actors/generate.mdx
Comment (connorsmith256) [inline docs/developer/actors/generate.mdx]: Suggestion: link to the concepts page on IDD

File: docs/developer/actors/generate.mdx
Comment (connorsmith256) [inline docs/developer/actors/generate.mdx]: Suggestion: link to Concepts page on actors

File: docs/developer/actors/build.mdx
Comment (connorsmith256) [inline docs/developer/actors/build.mdx]: Suggestion: link to Concepts page on actors

File: docs/developer/actors/publish.mdx
Comment (connorsmith256) [inline docs/developer/actors/publish.mdx]: Nit: I wouldn't use wasmcloud.azurecr.io as the example here, since it doesn't require credentials to pull artifacts

File: docs/developer/actors/run.mdx
Comment (connorsmith256) [inline docs/developer/actors/run.mdx]: Should we use "launch" or "start"?

File: docs/developer/actors/run.mdx
Comment (connorsmith256) [inline docs/developer/actors/run.mdx]:
```suggestion
This page took you through the process of imperatively starting an actor and a capability provider, and then linking them together. You can use these commands as you develop to work with running resources, or just use [wash app](/docs/cli/app) to deploy the same resources declaratively.
```

File: docs/developer/actors/test.mdx
Comment (connorsmith256) [inline docs/developer/actors/test.mdx]: Why "module"?

File: docs/developer/actors/test.mdx
Comment (connorsmith256) [inline docs/developer/actors/test.mdx]: s/modules/components?

File: docs/developer/actors/update.mdx
Comment (connorsmith256) [inline docs/developer/actors/update.mdx]: I've been slowly removing references to version numbers since they tend to get outdated. 0.18.0 is _so_ 2023!

Comment (connorsmith256) [inline docs/developer/actors/update.mdx]: Side note: IMO if we're going to rely on `wash dev` in our documentation we should remove the experimental flag from it

File: docs/developer/actors/update.mdx
Comment (connorsmith256) [inline docs/developer/actors/update.mdx]: s/module/component?

File: docs/reference/wasi/incoming-http.md
Comment (connorsmith256) [inline docs/reference/wasi/incoming-http.md:1]: Why is incoming-http its own page instead of with the other interfaces?

File: docs/reference/wasi/README
Comment (connorsmith256) [inline docs/reference/wasi/README:1]: Does this get rendered? I couldn't find the page in the preview site

File: docs/tour/hello-world.mdx
Comment (connorsmith256) [inline docs/tour/hello-world.mdx]: Should we link to https://github.com/wasmCloud/wasmCloud/blob/main/examples/docker/docker-compose-full.yml so we don't have to keep these in sync?

File: docs/tour/hello-world.mdx
Comment (connorsmith256) [inline docs/tour/hello-world.mdx:68]: Nit: I don't think new users will know the difference between "Rust" and "Rust Component"

File: docs/tour/hello-world.mdx
Comment (connorsmith256) [inline docs/tour/hello-world.mdx:200]: Nit: link to wadm docs instead of repo?

**Review by connorsmith256 — CHANGES_REQUESTED** [review]
🚀 just nits!

File: docs/developer/actors/test.mdx
Comment (brooksmtownsend) [reply docs/developer/actors/test.mdx]: Only because this guide was for the smithy-based actors. I don't know if there's an actual easy way to "test" the WASI HTTP stuff, so I might have to punt on this page altogether

File: docs/developer/actors/run.mdx
Comment (brooksmtownsend) [reply docs/developer/actors/run.mdx]: Great point, start is better

File: docs/developer/actors/update.mdx
Comment (brooksmtownsend) [reply docs/developer/actors/update.mdx]: For sure. I think wash dev should get support for wadm manifests and then we drop the experimental flag

File: docs/reference/wasi/README
Comment (brooksmtownsend) [reply docs/reference/wasi/README:1]: It does https://deploy-preview-250--dreamy-golick-5f201e.netlify.app/docs/reference/wasi/support

File: docs/reference/wasi/incoming-http.md
Comment (brooksmtownsend) [reply docs/reference/wasi/incoming-http.md:1]: We actually have different worlds for the guest things, https://github.com/wasmCloud/wasmCloud/blob/main/crates/runtime/wit/guest.wit, this is just the way it is as of 0.81 but not opposed to considering another organization method

File: docs/tour/hello-world.mdx
Comment (brooksmtownsend) [reply docs/tour/hello-world.mdx:68]: Do you think it's confusing enough to remove the old example entirely? I didn't like it either, and the difference between a module and a component in this case is really just the SDK you use to write the Wasm. I felt that it was odd to rework the docs to use the component examples and still mark them as experimental when it comes to possibly changing WASI APIs

File: docs/developer/actors/test.mdx
Comment (connorsmith256) [reply docs/developer/actors/test.mdx]: Ahh gotcha 👍

File: docs/developer/actors/update.mdx
Comment (connorsmith256) [reply docs/developer/actors/update.mdx]: We can resolve this for now, and we can come back to this after https://github.com/wasmCloud/wasmCloud/pull/1269 is released

File: docs/reference/wasi/README
Comment (connorsmith256) [reply docs/reference/wasi/README:1]: Isn't that the support page, and this is the README page?

File: docs/tour/hello-world.mdx
Comment (connorsmith256) [reply docs/tour/hello-world.mdx:68]: Yeah, we're in a bit of a bind for now. How about `Rust (Module)` and `Rust (Component)`?

File: docs/reference/wasi/README
Comment (brooksmtownsend) [reply docs/reference/wasi/README:1]: Oh! I see. I intentionally didn't want to render this page as it's more for docs maintainers, but if you think this content is relevant I can add the page

Comment (connorsmith256) [reply docs/reference/wasi/README:1]: Ahh, that makes sense 👍

File: docs/tour/hello-world.mdx
Comment (lachieh) [reply docs/tour/hello-world.mdx:68]: If you wanted to formalize the use of these, you could create a component that re-exports the tabs with no access to change the props. Usage example below:

```mdx
<LangTabs>
  <RustTab>...</RustTab>
  <RustTab component>...</RustTab>
  <GoTab>...</GoTab>
  <JSTab>...</JSTab>
  <PythonTab>...</PythonTab>
  ...etc
</LangTabs>
```

**Review by lachieh — APPROVED** [review]
Love it. Re-org of actions build/run/etc is nice and I love the generated docs.

File: docs/tour/hello-world.mdx
Comment (lachieh) [reply docs/tour/hello-world.mdx:68]: If you want to do this, let me know and I can assist. There's a way to export components so they don't need to be imported in every file, too.

Comment (brooksmtownsend) [reply docs/tour/hello-world.mdx:68]: I don't see it as a blocker for this PR but I'd love to implement this. Same thing with OS/platform things. @lachieh would you mind filing an enhancement issue?

Comment (lachieh) [reply docs/tour/hello-world.mdx:68]: [Done](https://github.com/wasmCloud/wasmcloud.com/issues/254)

**Comment by connorsmith256** [comment]
@brooksmtownsend I can resolve the conflict and merge this if you'd like. If I don't hear from you by EOD I'll assume you're having a nice time off and will take care of it :)

**Comment by brooksmtownsend** [comment]
@connorsmith256 that would be wonderful 💗 thank you. From my perspective this is good to go once the other linked PR merges!

**Review by connorsmith256 — APPROVED** [review]

**Pull request merged**
